### PR TITLE
Functional tests - Product settings enable/disable default activation status

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | enable/disable default activation status
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/03_productSettings/06_enableDisableDefaultActivationStatus" URL_FO=yourShopUrl npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17605)

### CODE REVIEW

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js [hunk 1]
```diff
@@ -53,4 +54,15 @@ module.exports = class productSettings extends BOBasePage {
     await this.clickAndWaitForNavigation(this.saveProductGeneralFormButton);
     return this.getTextContent(this.alertSuccessBloc);
   }
+
+  /**
+   * Change default activation status
+   * @param toEnable
+   * @returns {Promise<string|*>}
+   */
+  async changeDefaultActivationStatus(toEnable = true) {
```
Comment (Progi1984) [inline tests/puppeteer/pages/BO/shopParameters/productSettings/index.js]:
```suggestion
  async setDefaultActivationStatus(toEnable = true) {
```

**Review by Progi1984 — COMMENTED** [review]
A small comment

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@nesrineabdmouleh Please fix conflicts :)

File: tests/puppeteer/pages/BO/catalog/products/add.js [hunk 2]
```diff
@@ -21,6 +21,8 @@ module.exports = class AddProduct extends BOBasePage {
     this.saveProductButton = 'input#submit[value=\'Save\']';
     this.previewProductLink = 'a#product_form_preview_btn';
     this.productOnlineSwitch = '.product-footer div.switch-input';
+    this.productOnlineTitle = 'h2.for-switch.online-title';
+    this.productDescriotionTab = '#tab_description a';
```
Comment (Progi1984) [inline tests/puppeteer/pages/BO/catalog/products/add.js]:
```suggestion
    this.productDescriptionTab = '#tab_description a';
```

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js [hunk 3]
```diff
@@ -77,4 +78,15 @@ module.exports = class productSettings extends BOBasePage {
...
+
+  /**
+   * Change default activation status
+   * @param toEnable
+   * @returns {Promise<string|*>}
+   */
+  async setDefaultActivationStatus(toEnable = true) {
+    await this.waitForSelectorAndClick(this.switchDefaultActivationStatusLabel.replace('%TOGGLE', toEnable ? 1 : 0));
+    await this.clickAndWaitForNavigation(this.saveProductGeneralFormButton);
+    return this.getTextContent(this.alertSuccessBloc);
```
Comment (Progi1984) [inline tests/puppeteer/pages/BO/shopParameters/productSettings/index.js:90]:
```suggestion
    return this.getTextContent(this.alertSuccessBlock);
```

**Review by Progi1984 — COMMENTED** [review]
Some feedback

File: tests/puppeteer/pages/BO/catalog/products/add.js [hunk 4]
```diff
@@ -295,4 +297,12 @@ module.exports = class AddProduct extends BOBasePage {
     await this.goToFormStep(5);
     return this.getAttributeContent(this.friendlyUrlInput, 'value');
   }
+
+  /**
+   * Get online product status
+   * @returns {Promise<boolean>}
+   */
+  async getOnlineButtonStatus() {
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/BO/catalog/products/add.js]:
```suggestion
  getOnlineButtonStatus() {
```
No need to async if there is no await instruction

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js (hunk 3 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/pages/BO/shopParameters/productSettings/index.js:90]: The selector name is used in many other files, maybe I should create another PR to fix the name in all files. 
Wdyt?

Comment (Progi1984) [reply tests/puppeteer/pages/BO/shopParameters/productSettings/index.js:90]: :ok_hand: for an another PR

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**